File: docs/snippets/all/archetypes/line_strip2d_ui_radius.rs
//! Log lines with ui points & scene unit radii.

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let rec = rerun::RecordingStreamBuilder::new("rerun_example_line_strip2d_ui_radius").spawn()?;

    // A blue line with a scene unit radii of 0.01.
    let points = [[0., 0.], [0., 1.], [1., 0.], [1., 1.]];
    rec.log(
        "scene_unit_line",
        &rerun::LineStrips2D::new([points])
            // By default, radii are interpreted as world-space units.
            .with_radii([0.01])
            .with_colors([rerun::Color::from_rgb(0, 0, 255)]),
    )?;

    // A red line with a ui point radii of 5.
    // Ui points are independent of zooming in Views, but are sensitive to the application ui scaling.
    // For 100 % ui scaling, ui points are equal to pixels.
    let points = [[3., 0.], [3., 1.], [4., 0.], [4., 1.]];
    rec.log(
        "ui_points_line",
        &rerun::LineStrips2D::new([points])
            // rerun::Radius::new_ui_points produces a radius that the viewer interprets as given in ui points.
            .with_radii([rerun::Radius::new_ui_points(5.0)])
            .with_colors([rerun::Color::from_rgb(255, 0, 0)]),
    )?;

    // TODO(#5520): log VisualBounds2D

    Ok(())
}

File: docs/snippets/all/archetypes/line_strip2d_ui_radius.py
"""Log lines with ui points & scene unit radii."""

import rerun as rr
import rerun.blueprint as rrb

rr.init("rerun_example_line_strip2d_ui_radius", spawn=True)

# A blue line with a scene unit radii of 0.01.
points = [[0, 0], [0, 1], [1, 0], [1, 1]]
rr.log(
    "scene_unit_line",
    rr.LineStrips2D(
        [points],
        # By default, radii are interpreted as world-space units.
        radii=0.01,
        colors=[0, 0, 255],
    ),
)

# A red line with a ui point radii of 5.
# UI points are independent of zooming in Views, but are sensitive to the application UI scaling.
# For 100% ui scaling, UI points are equal to pixels.
points = [[3, 0], [3, 1], [4, 0], [4, 1]]
rr.log(
    "ui_points_line",
    rr.LineStrips2D(
        [points],
        # rr.Radius.ui_points produces radii that the viewer interprets as given in ui points.
        radii=rr.Radius.ui_points(5.0),
        colors=[255, 0, 0],
    ),
)

# Set view bounds:
rr.send_blueprint(rrb.Spatial2DView(visual_bounds=rrb.VisualBounds2D(x_range=[-1, 5], y_range=[-1, 2])))
